src/site/_DevSiteRouter.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {
+  useLocation, useRouteMatch, Redirect,
+} from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { sitesPropType, routesMapPropType } from './siteConfigPropTypes';
+import getSessionStorage from '../browser-router-redirect/session';
+
+const propTypes = {
+  /**
+   * The site config for the application.
+   */
+  sites: sitesPropType.isRequired,
+  /**
+   * The map of routes to redirect to
+   */
+  routesMap: routesMapPropType.isRequired,
+  /**
+   * Children
+   */
+  children: PropTypes.node.isRequired,
+};
+
+const DevSiteRouter = ({
```

Comment:
This class does all the awkward routing for us, routing the root to the first content item, routing /primaryNavItem to the first content item of that type and flipping to other sites based on their reserved routes.